test/JDBC/expected/non_default_server_collation/chinese_prc_ci_as/babel_collection.out

```diff
@@ -361,14 +372,16 @@ GO
 ~~START~~
 text
 Query Text: select c1 from testing4 where c1 LIKE '\%ones'
-Bitmap Heap Scan on testing4  (cost=11.40..32.78 rows=1 width=32)
-  Filter: (((c1)::text ~~* '\%ones'::text COLLATE chinese_prc_cs_as) AND ((c1)::text >= '\'::text) AND ((c1)::text < '\?'::text))
-  ->  Bitmap Index Scan on c1_idxtesting49a168d73f3ba5aacdfd495b931b8d187  (cost=0.00..11.40 rows=650 width=0)
+Bitmap Heap Scan on testing4  (cost=4.18..11.30 rows=1 width=32)
+  Recheck Cond: ((c1 >= '\'::"varchar") AND (c1 < '\?'::"varchar"))
+  Filter: ((c1)::text ~~* '\%ones'::text COLLATE chinese_prc_cs_as)
+  ->  Bitmap Index Scan on c1_idxtesting49a168d73f3ba5aacdfd495b931b8d187  (cost=0.00..4.18 rows=3 width=0)
+        Index Cond: ((c1 >= '\'::"varchar") AND (c1 < '\?'::"varchar"))
```

Comment:
why is transformation applied here? because pattern is not const prefix

Reply:
This is a const node, PG also shows optimisation